applications/wg-easy/Taskfile.yaml

```diff
@@ -519,3 +528,99 @@ tasks:
       - task: helm-install
       - task: test
       - task: cluster-delete
+
+  cmx-vm-create:
+    desc: Create a CMX VM instance using Replicated CLI
+    run: once
+    silent: false
+    status:
+      - |
+        # Check if VM is running
+        replicated vm ls | grep "{{.CMX_VM_NAME}}" | grep running
+    cmds:
+      - |
+        echo "Creating CMX VM {{.CMX_VM_NAME}}..."
+        replicated vm create --distribution {{.CMX_VM_DISTRIBUTION}} --version {{.CMX_VM_VERSION}} --instance-type {{.CMX_VM_INSTANCE_TYPE}} --disk {{.CMX_VM_DISK_SIZE}} --name {{.CMX_VM_NAME}} --ttl {{.CMX_VM_TTL}}
+
+        echo "Waiting for VM to be running (timeout: 120s)..."
+        for i in $(seq 1 60); do
+          replicated vm ls | grep "{{.CMX_VM_NAME}}"
+          if replicated vm ls | grep "{{.CMX_VM_NAME}}" | grep running; then
+            echo "VM {{.CMX_VM_NAME}} is ready!"
+            break
+          fi
+          if [ $i -eq 60 ]; then
+            echo "Timeout reached after 120s. VM may still be initializing."
+            exit 1
+          fi
+          sleep 2
+        done
+
+  cmx-vm-delete:
+    desc: Delete a CMX VM instance
+    silent: false
+    cmds:
+      - |
+        echo "Deleting CMX VM {{.CMX_VM_NAME}}..."
+        replicated vm rm {{.CMX_VM_NAME}}
+
+  cmx-vm-install:
+    desc: Download and install the app as Embedded Cluster on CMX VM
+    silent: false
+    requires:
+      vars: [REPLICATED_LICENSE_ID]
+    vars:
+      CHANNEL: '{{.CHANNEL | default "Unstable"}}'
+      SKIP_INSTALL: '{{.SKIP_INSTALL | default "false"}}'
+      ADMIN_CONSOLE_PASSWORD:
+        sh: uuidgen | cut -c1-13
+    deps:
+      - cmx-vm-create
+    status:
+      - replicated vm port ls {{.CMX_VM_NAME}}
+      - curl -sL $(replicated vm port ls {{.CMX_VM_NAME}} --output=json | jq -r ".[0].hostname")/healthz
+    cmds:
+      - |
+        echo "Check if user is set..."
+        if [ -z "{{.CMX_VM_USER}}" ]; then
+          echo "CMX_VM_USER is not set. Set it to your Github handle. E.g. task cmx-vm-install CMX_VM_USER=nvanthao"
+          exit 1
+        fi
+
+        echo "Check if license ID is set..."
+        if [ -z "{{.REPLICATED_LICENSE_ID}}" ]; then
+          echo "REPLICATED_LICENSE_ID is not set. Set it to your DEV Customer License ID. E.g. task cmx-vm-install REPLICATED_LICENSE_ID=1234567890"
+          exit 1
+        fi
+
+        echo "SSH into the VM and download the app binary..."
+        SSH_BASE_CMD="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
+        if [ -n "{{.CMX_VM_PUBLIC_KEY}}" ]; then
+          SSH_BASE_CMD="$SSH_BASE_CMD -i {{.CMX_VM_PUBLIC_KEY}}"
+        fi
+        VM_SSH_CMD=$(replicated vm ls --output=json | jq -r ".[] | select(.name == \"{{.CMX_VM_NAME}}\") | \"$SSH_BASE_CMD -p \(.direct_ssh_port) {{.CMX_VM_USER}}@\(.direct_ssh_endpoint)\"")
```

Comment:
For future travellers. Using `replicated vm ssh-endpoint` does not yet work if your replicated token is a service token. Prefer this `vm ls | jq ...` oneliner

Reply:
I tagged @chris-sanders to the Slack thread discussion on this.